### PR TITLE
add gateway server name as stat prefix

### PR DESCRIPTION
**Pull request by ramaraochavali**
Add gateway server name as stat prefix for http connection manager if available
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@rshriram PTAL

File: pilot/pkg/networking/core/v1alpha3/gateway.go [hunk 1]
```diff
@@ -398,6 +391,8 @@ func (configgen *ConfigGeneratorImpl) createGatewayHTTPFilterChainOpts(
 				ServerName:          EnvoyServerName,
 				HttpProtocolOptions: httpProtoOpts,
 			},
+			addGRPCWebFilter: serverProto == protocol.GRPCWeb,
```
Comment (ramaraochavali) [inline pilot/pkg/networking/core/v1alpha3/gateway.go:394]: For Http, we seems to use a single http connection manager - So I am not setting the stat prefix there.

**Comment by howardjohn** [comment]
temporarily putting a hold for reasons unrelated to this PR (test infra issue).

**Comment by howardjohn** [comment]
removed hold, fixed the issue

File: pilot/pkg/model/gateway.go [hunk 2]
```diff
@@ -84,12 +85,21 @@ func MergeGateways(gateways ...Config) *MergedGateway {
...
+		gatewayName := gatewayConfig.Namespace + "/" + gatewayConfig.Name // Format: %s/%s
 		names[gatewayName] = true
 
 		gatewayCfg := gatewayConfig.Spec.(*networking.Gateway)
 		log.Debugf("MergeGateways: merging gateway %q into %v:\n%v", gatewayName, names, gatewayCfg)
+		snames := sets.Set{}
 		for _, s := range gatewayCfg.Servers {
+			if len(s.Name) > 0 {
+				if snames.Contains(s.Name) {
+					log.Warnf("Server name %s is not unique in gateway %s and may create possible issues like stat prefix collision ",
```
Comment (howardjohn) [inline pilot/pkg/model/gateway.go:97]: isn't this going to be super spammy? This `name` thing is a new field right? So 100% of users have it unset, meaning that anyone with 2+ gateways will get this every time this codepath is executed, for every gateway, at warn level. I think this code is called pretty often too

Comment (ramaraochavali) [reply pilot/pkg/model/gateway.go:97]: Should n't this help if len(s.Name) > 0?

Comment (howardjohn) [reply pilot/pkg/model/gateway.go:97]: Yep, thanks! missed that

**Pull request merged**